src/files/actions/approveAction.js

```diff
@@ -0,0 +1,31 @@
+import CheckCircleSvgIcon from '@mdi/svg/svg/check-circle.svg'
+import { Permission, FileAction } from '@nextcloud/files'
+import { states } from '../../states.js'
+import { onApproveAction } from '../helpers.js'
+
+export const approveAction = new FileAction({
+	id: 'approval-approve',
+	displayName: (nodes) => {
+		return t('approval', 'Approve')
+	},
+	enabled(nodes, view) {
+		return !OCA.Approval.actionIgnoreLists.includes(view.id)
+			&& !nodes.some(({ permissions }) => (permissions & Permission.READ) === 0)
+			&& nodes.some(node => node.attributes['approval-state'] === states.APPROVABLE)
+		// && nodes.every(({ type }) => type === FileType.File)
+		// && nodes.every(({ mime }) => mime === 'application/some+type')
+	},
+	iconSvgInline: () => CheckCircleSvgIcon,
+	async exec(node) {
+		onApproveAction(node)
+		return null
+	},
+	async execBatch(nodes) {
+		nodes
+			.filter(node => node.attributes['approval-state'] === states.APPROVABLE)
+			.forEach(node => {
+				onApproveAction(node)
+			})
+		return nodes.map(_ => null)
```

Comment:
This will makes the selection stay smart and keep relevant files in case some failed to be approved.

```suggestion
		const promises = nodes
			.filter(node => node.attributes['approval-state'] === states.APPROVABLE)
			.map(onApproveAction)
		const results = await Promise.allSettled(promises)
		return results.map(promise => promise.status === 'fulfilled')
```

Reply:
As the error messages are handled somewhere else, I think it does not matter if we wait or not, we don't wanna know anything anymore once we called `onApproveAction`. 
I still rewrote everything with async/await and exec/execBatch are now waiting for all promises to settle.

Reply:
It does, the selection is smart enough to keep failed items for the user convenience. See new review